src/gqlRules/gqlNameMatch/gqlNameMatch.js

```diff
@@ -0,0 +1,95 @@
+import { isGqlFile } from "../utils";
+import { isQuery, isMutation } from "../utils/isGQLOperation";
+import { relativePathToFile } from "../../utils";
+import { operationName } from "../utils/operationName";
+import { sanitizeGqlOperationText } from "../utils/sanitizeGqlOperationText";
+
+const meta = {
+  type: "problem",
+  docs: {
+    description:
+      "The Variable name GQL operation is assigned should match with the GQL operation name Ex: const Patient_Query_UserAvatar = gql` query Patient_Query_UserAvatar...`",
+    category: "gql-name-match",
+    recommended: false,
+  },
+  fixable: "code",
+  schema: [
+    {
+      type: "object",
+      properties: {
+        namespaceIgnoreList: {
+          type: "array",
+          items: {
+            type: "string",
+          },
+        },
+        namespaceOperationPrefix: {
+          type: "object",
+        },
+      },
+      required: ["namespaceOperationPrefix"],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const create = context => {
+  const isGqlObjectFile = isGqlFile(context);
+  const { namespaceOperationPrefix, namespaceIgnoreList } = context.options[0];
+
+  const isGqlTemplateElement = node => {
+    return node.tag && node.tag.name === "gql";
+  };
+
+  const getNamespace = () => {
+    const pathToFile = relativePathToFile(context);
+    const relativeJsPath = pathToFile.replace("app/javascript/", "");
+
+    const namespace = Object.keys(namespaceOperationPrefix).find(namespaceOperationKey =>
+      relativeJsPath.startsWith(namespaceOperationKey)
+    );
+    return { namespace };
+  };
+
+  const isInIgnoreList = namespace => {
+    return namespaceIgnoreList.includes(namespace);
+  };
+
+  const validateOperationName = (gqlOperationText, operationType, node) => {
+    const gqlOperationName = operationName(gqlOperationText, operationType);
+    const { namespace } = getNamespace();
+    const { id } = node;
+    const variableName = id.name;
+    if (isInIgnoreList(namespace)) {
+      return;
+    }
+
+    const isOperationNameValid = !!gqlOperationName.match(variableName);
+
+    if (!isOperationNameValid) {
+      context.report({
+        node: node,
+        message: `The variable name "${variableName}" should match with the GQL operation name, please use "${gqlOperationName}"`,
+      });
+    }
+  };
+
+  const VariableDeclarator = node => {
+    if (!isGqlObjectFile || !isGqlTemplateElement(node.init)) return;
+    const { init } = node;
+    const templateElementNode = init.quasi;
+    const gqlOperationText = sanitizeGqlOperationText(templateElementNode, context);
+
+    if (isQuery(gqlOperationText)) {
+      validateOperationName(gqlOperationText, "Query", node);
+    } else if (isMutation(gqlOperationText)) {
+      validateOperationName(gqlOperationText, "Mutation", node);
+    }
```

Comment:
Since we don't need the namespace stuff anymore and we aren't validating the name format, we can refactor this such that we don't have logic checking if it's a mutation or query etc. 

This whole block:
```ts
   if (isQuery(gqlOperationText)) {
      validateOperationName(gqlOperationText, "Query", node);
    } else if (isMutation(gqlOperationText)) {
      validateOperationName(gqlOperationText, "Mutation", node);
    }
```

becomes just:
```ts
      validateOperationName(gqlOperationText, node);
```

Make sure to also change the parameters of `validateOperationName` to remove the middle parameter (operationType).

Reply:
I noticed that `operationName` util function is referring to the `Query` and `Mutation` text and getting the operationName based on that text so we still have to check whether if the operation is query or mutation.

I created a variable for operationType assuming it's safe to do so since there are only two types of operations and we're returning if the file is not gql object file before the `IsOperationNameAndVariableNameSame` function is called.

`const operationType = isQuery(gqlOperationText) ? "Query" : "Mutation"`